skbio/stats/composition.py

```diff
@@ -806,6 +807,163 @@ def centralize(mat):
+       a 1-dimensional vector of proportions
+    y : array_like, float
+       a 1-dimensional vector of proportions
+        
+    ddof: int
+        degrees of freedom
+        
+    robust: bool
+        mask zeros at the cost of performance
+       
+    Returns
+    -------
+    float
+         variance log ratio value
+         
+    Examples
+    --------
+    # No zeros
+    >>> x = [1,2,3]
+    >>> y = [5,8,13]
+    >>> %timeit vlr(x,y)
+    # 21.7 µs ± 979 ns per loop (mean ± std. dev. of 7 runs, 10000 loops each)
+    # 0.01277962183258352
+
+    # Zeros without robust
+    >>> x = [1,2,3,0]
+    >>> y = [5,8,13,21]
+    >>> %timeit vlr(x,y)
+    # 27.5 µs ± 2.48 µs per loop (mean ± std. dev. of 7 runs, 10000 loops each)
+    # nan
+
+    # Zeros with robust
+    >>> x = [1,2,3,0]
+    >>> y = [5,8,13,21]
+    >>> %timeit vlr(x,y, robust=True)
+    # 570 µs ± 21 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)
+    # 0.01277962183258352
+    
+    References
+    ----------
+    .. [1] V. Lovell D, Pawlowsky-Glahn V, Egozcue JJ, Marguerat S, Bähler J (2015) 
+           Proportionality: A Valid Alternative to Correlation for Relative Data. 
+           PLoS Comput Biol 11(3): e1004075. https://doi.org/10.1371/journal.pcbi.1004075
+    .. [2] Erb, I., Notredame, C. 
+           How should we measure proportionality on relative gene expression data?. 
+           Theory Biosci. 135, 21–36 (2016). https://doi.org/10.1007/s12064-015-0220-8
+    """
+    # Convert array_like to numpy array
+    x = np.asarray(x)
+    y = np.asarray(y)
+
+    # Mask zeros
+    if robust:
+        x = np.ma.masked_array(x, mask=x == 0)
+        y = np.ma.masked_array(y, mask=y == 0)
+    
+    # Log transformation
+    x = np.log(x)
+    y = np.log(y)
+    
+    #Variance log ratio
+    return np.var(x - y, ddof=ddof)
+
+@experimental(as_of="0.5.7")
+def pairwise_vlr(mat:Union[np.array, pd.DataFrame], ids:list=None, ddof:int=1, robust=False) -> DistanceMatrix:
+    r"""
+    Performs pairwise variance log ratio transformation.
+    
+    Parameters
+    ----------
+    mat : array_like, float
+       a matrix of proportions where
+       rows = compositions and
+       columns = components
+       
+    ids: array_like, str
+        component names
+        if pd.DataFrame is provided and `ids is None` then `mat.columns` are used for `ids`
+        
+    ddof: int
+        degrees of freedom
+        
+    robust: bool
+        mask zeros at the cost of performance
+       
+    Returns
+    -------
+    skbio.DistanceMatrix
+         distance matrix of variance log ratio values
+    Examples
+    --------
+    >>> # NumPy Array
+    >>> mat = np.asarray([
+        [1,2,3],
+        [5,8,13],
+        [21,34,55],
+    ])
+    >>> dism = pairwise_vlr(mat)
+    >>> dism.redundant_form()
+    # array([[0.        , 0.01576411, 0.00649553],
+    #        [0.01576411, 0.        , 0.00202147],
+    #        [0.00649553, 0.00202147, 0.        ]])
```

Comment:
I don't think this is going to pass the doc tests -- the doc tests won't like to have comments in it.
So this should be instead
```python
     array([[0.        , 0.01576411, 0.00649553],
            [0.01576411, 0.        , 0.00202147],
            [0.00649553, 0.00202147, 0.        ]])
```